site/app/controllers/forum/ForumController.php

```diff
@@ -401,12 +404,26 @@ public function alterPost($modifyType){
                 $thread_id = $_POST["thread_id"];
                 $post_id = $_POST["post_id"];
                 $type = "";
-                if($this->core->getQueries()->deletePost($post_id, $thread_id)){
+                if($this->core->getQueries()->setDeletePostStatus($post_id, $thread_id, 1)){
                     $type = "thread";
                 } else {
                     $type = "post";
                 }
                 $this->core->getOutput()->renderJson(array('type' => $type));
+            } else if($modifyType == 2) { //undelete post or thread
```

Comment:
Lets add some comments above the method declaration, similar to your comments for setDeletePostStatus. It might be worth breaking this function up in the future.